.github/workflows/build.yaml

```diff
@@ -192,12 +321,121 @@ jobs:
+          retention-days: 5
+
+  build_3rd_party_dependencies:
+    name: Build 3rd Party Dependencies
+    needs: [workflow-setup, indy_plenum_tests, indy_plenum_module_tests, lint]
+    runs-on: ubuntu-20.04
+    env:
+      UBUNTU_VERSION: ${{ needs.workflow-setup.outputs.UBUNTU_VERSION }}
+    container:
+      image: ghcr.io/${{ needs.workflow-setup.outputs.GITHUB_REPOSITORY_NAME }}/plenum-build:${{ needs.workflow-setup.outputs.UBUNTU_VERSION }}
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v1
+
+      - name: Try load from cache.
+        id: third-party-dependencies
+        uses: actions/cache@v2
+        with:
+          path: /tmp/third-party-dependencies
+          key: ${{ format('third-party-dependencies-{0}', hashFiles(format('./build-scripts/{0}/build-3rd-parties.sh', needs.workflow-setup.outputs.UBUNTU_VERSION ))) }}           
+   
+      - name: Build 3rd party deployment packages
+        if: steps.third-party-dependencies.outputs.cache-hit != 'true'
+        run: |
+          mkdir -p ./build-scripts/${{ needs.workflow-setup.outputs.UBUNTU_VERSION }}/cache/3rd-party-dependencies/
+          ./build-scripts/${{ needs.workflow-setup.outputs.UBUNTU_VERSION }}/build-3rd-parties.sh ./cache/3rd-party-dependencies
+          mv ./build-scripts/${{ needs.workflow-setup.outputs.UBUNTU_VERSION }}/cache/* /tmp/third-party-dependencies
+
+  publish_artifacts:
+    name: Publish Artifacts
+    runs-on: ubuntu-20.04
+    needs: [workflow-setup, build_release, build_3rd_party_dependencies]
+    if: needs.workflow-setup.outputs.publish == 'true'
+    env:
+      GITHUB_REF: ${{ needs.workflow-setup.outputs.GITHUB_REF }}
+      UBUNTU_VERSION: ${{ needs.workflow-setup.outputs.UBUNTU_VERSION }}
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v1
+
+      - name: Setup JFrog CLI
+        uses: jfrog/setup-jfrog-cli@v1
+        env:
+          JF_ARTIFACTORY_1: ${{ secrets.INDY_ARTIFACTORY_REPO_CONFIG }}
+
+      - name: Ping Artifactory
+        run: |
+          # Test the connection to Ping the Hyperledger Artifactory server
+          # to ensure everything has been setup correctly.
+          jfrog rt ping
+
+      - name: Download Plenum Artifacts from Pipeline Artifacts
+        uses: actions/download-artifact@v2
+        with:
+          name: plenum-deb
+          path: to_publish
+
+      - name: Publish Plenum Artifacts
+        uses: ./.github/actions/publish-deb
+        with:
+          sourceDirectory: /home/runner/work/indy-plenum/indy-plenum/to_publish
+          distribution: focal
+          component: ${{ env.GITHUB_REF }}
+
+      - name: Download 3rd Party Artifacts Dependencies from Cache
+        id: third-party-dependencies
+        uses: actions/cache@v2
+        with:
+          path: /tmp/third-party-dependencies
+          key: third-party-dependencies-${{ hashFiles('./build-scripts/${{ env.UBUNTU_VERSION }}/build-3rd-parties.sh') }}
+          restore-keys: |
```

Comment:
@WadeBarnes. I think we should remove the `restore-keys` here as we want to explicitly use the cache from `build_3rd_party_dependencies` if the hash value of `build-3rd-parties.sh` changes.
What do you think?

Reply:
Sounds good.